### PR TITLE
Implement library for Problems (RFC7807)

### PR DESCRIPTION
**Pull request by Ekberg**
This PR implements a library for using [RFC 7807](https://tools.ietf.org/html/rfc7807)-based error responses.

A common interface for all Problems as well as a few common implementations of generic Problems useful for validating and returning internal errors.

### CODE REVIEW

File: problems/http.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+		Status: status,
+	}
+}
+
+// WriteResponse, converts the error into a Problem and writes the contents into w.
+// The Problem will be decorated with request information and logged if possible.
+func WriteResponse(ctx context.Context, err error, w http.ResponseWriter, r *http.Request) {
+	problem := FromError(err)
+
+	if decoratableProblem, ok := problem.(RequestDecoratableProblem); ok {
```
Comment (rickardenglund) [inline problems/http.go:24]: Verify that this works for validationProblem

File: problems/basic.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+	// This should always be the same value for the same Type.
+	Title string `json:"title"`
+
+	// Status, the HTTP status code associated with this problem occurrence.
+	Status int `json:"status,omitempty"`
+
+	// Detail, a human-readable explanation specific to this occurrence of the problem.
+	Detail string `json:"detail,omitempty"`
+
+	// Instance, a URI reference that identifies the specific occurrence of the problem.
```
Comment (j0s) [inline problems/basic.go:31]: Should this be "identifies the specific resource on which the problem occurred"?

File: problems/validation.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+func (problem ValidationProblem) TrimEmpty() error {
+	if len(problem.Reasons) == 0 {
+		return nil
+	}
+
+	return problem
+}
+
+func (problem *ValidationProblem) Errors() []error {
+	var errors []error = nil //nolint:prealloc // expected behaviour is to return nil when no reasons is present
```
Comment (rickardenglund) [inline problems/validation.go:31]: This looks like a named parameter

**Review by rickardenglund — APPROVED** [review]

File: problems/http.go (hunk 1 above)
Comment (j0s) [reply problems/http.go:24]: This works as intdended

File: problems/validation.go (hunk 3 above)
Comment (j0s) [reply problems/validation.go:31]: yup

**Review by j0s — DISMISSED** [review]

**Review by j0s — APPROVED** [review]

**Pull request merged**